api/suggestion/controllers/suggestion.js

```diff
@@ -7,6 +7,77 @@ const { parseMultipartData, sanitizeEntity } = require("strapi-utils");
  */
 
 module.exports = {
+  /**
+   * Endpoint for creating new suggestions. This saves the suggestions in draft state,
+   * so that they can be later published by admins.
+   */
+  async new(ctx) {
+    let entity, data, files;
+
+    if (ctx.is("multipart")) {
+      ({ data, files } = parseMultipartData(ctx));
+    } else {
+      data = ctx.request.body;
+    }
+
+    // Set published_at null so the entry will be created in draft state
+    data.published_at = null;
+    data.from_web = true;
+
+    entity = await strapi.services.suggestion.create(
+      data,
+      files ? { files } : undefined
+    );
+
+    return sanitizeEntity(entity, { model: strapi.models.suggestion });
+  },
+  /**
+   * Endpoint for liking suggestions.
+   */
+  async like(ctx) {
+    const { id } = ctx.params;
+    const { user } = ctx.request.body;
+
+    let entity = await strapi.services.suggestion.findOne({ id });
+
+    if (!entity) {
+      ctx.response.status = 404;
+      return;
+    }
+
+    if (entity.likes?.includes(user)) {
+      return sanitizeEntity(entity, { model: strapi.models.suggestion });
+    }
+
+    entity.likes = [...(entity.likes || []), user];
+
+    entity = await strapi.services.suggestion.update({ id }, entity);
+
+    return sanitizeEntity(entity, { model: strapi.models.suggestion });
+  },
+  /**
+   * Endpoint for unliking suggestions.
+   */
+  async unlike(ctx) {
+    const { id } = ctx.params;
+    const { user } = ctx.request.body;
+
+    let entity = await strapi.services.suggestion.findOne({ id });
+
+    if (!entity) {
+      ctx.response.status = 404;
+      return;
+    }
+
+    const index = entity.likes?.findIndex((x) => x === user);
+
+    if (index > -1) {
+      entity.likes.splice(index, 1);
+      entity = await strapi.services.suggestion.update({ id }, entity);
+    }
+
+    return sanitizeEntity(entity, { model: strapi.models.suggestion });
+  },
```

Comment:
There probably should also be an error returned if the user tries to unlike something he hasnt liked before.
Below code hasn't been tested, just an idea:
```suggestion
    const entity = await strapi.services.suggestion.findOne({ id });

    if (!entity || !entity.likes?.includes(user)) {
      ctx.response.status = 404;
      return;
    }
    const filteredLikes = entity.likes?.filter((x) => x !== user)
    const updatedEntity = await strapi.services.suggestion.update({ id }, {...entity, likes: filteredLikes});'
    return sanitizeEntity(updatedEntity, { model: strapi.models.suggestion });
  },
```